### DISCUSSION TITLE
What is the main difference between source-tracking and sgd

### DISCUSSION THREAD

**Discussion by AllanOricil** [scolladon/sfdx-git-delta#598] [General]
Hi @scolladon could you describe what are the pros and cons of sgd when compared to sfdx source tracking
 https://github.com/forcedotcom/source-tracking

**Comment by scolladon** [comment]
Hi @AllanOricil, I don't know closely what does the source-tracking tool.
It seems to compare the local content with the current default org content for a sfdx project.
`sfdx-git-delta` is not tight to a particular org. It compares sources located on git.

I'm not sure to be able to list pros and cons of both tools without a specific scenario.
Why are you thinking about source-tracking ? Do you have a scenario in mind ?

**Reply by AllanOricil** [reply]
I was just thinking if they were both doing the same thing.

**Reply by AllanOricil** [reply]
It tracks the state of an instance and it is able to deploy the changes in it

**Reply by AllanOricil** [reply]
Can't it be used in CI/CD pipelines to enable delta deployments?

**Reply by scolladon** [reply]
I'm not sure, the lib seems to have a similar approach to detect changes.
They have a [`shadowRepo`](https://github.com/forcedotcom/source-tracking/blob/main/src/shared/localShadowRepo.ts#L117) 
 concept where they store changes done locally and then compare what have changed based on that. But they are managing this repo state.

It does not seems to compare based on the user repository.
They do similar things, interact with a git repository, create sources to be deployed and deleted.

We have a refactoring item to use SDR as a metadata repository (instead of our json files).
We could also use the metadata resolver, but I'm not sure it would fit the need of comparing old pointers together (as it is based on the current repo state).
The reason why the plugin does not use those yet is because the plugin is older than those sfdx libraries.
We looked to it, but it seems too fresh/new at first to migrate to it. Now we feel it is a better time to consider using SDR.
We will consider it after the typescript migration.

I hope it answers you question @AllanOricil

**Reply by AllanOricil** [reply]
What is this SDR?

**Reply by scolladon** [reply]
Yep sorry, it is mentioned in the source tracking repository but nothing points to it
Here is a [blog](https://developer.salesforce.com/blogs/2022/02/we-broke-deploy-retrieve-hooks) explaining and referencing elements to dive deeper if you feel you need to

**Comment by mehdicherf** [comment]
Hi @AllanOricil and @scolladon,

My 2 cents:

Source tracking and SGD are different tools that do not serve the same purpose.

- Source tracking is here to help you retrieve and commit locally any change you applied to a development org (which could be a Scratch org, a Developer Sandbox, etc.). For example. if you need to create an object in an org, and then commit that to the repo, the easiest way to do that is to do it an org with source tracking enabled, and then pull your changes locally before committing them to the git repo. Source tracking is used mainly by Developers or Business Analysts (that need to commit their changes to a repo).

- SGD helps later in the process, when you have a git repo which contains one or several changes, that you want to deploy to a higher org. SGD is used mainly within CI/CD pipelines (or manually by Release Managers / Tech Leads / Architects...)
